solaris/SolarisProcessList.c

```diff
@@ -50,13 +52,18 @@ ProcessList* ProcessList_new(UsersTable* usersTable, Hashtable* pidMatchList, ui
 
    spl->kd = kstat_open();
 
-   pl->cpuCount = sysconf(_SC_NPROCESSORS_ONLN);
+   pageSize = sysconf(_SC_PAGESIZE);
+   if (pageSize == -1)
+      CRT_fatalError("Cannot get pagesize by sysconf(_SC_PAGESIZE)");
+   pageSizeKB = pageSize / 1024;
 
-   if (pl->cpuCount == 1 ) {
+   pl->cpuCount = sysconf(_SC_NPROCESSORS_ONLN);
+   if (pl->cpuCount == -1)
+      CRT_fatalError("Cannot get CPU count by sysconf(_SC_NPROCESSORS_ONLN)");
+   else if (pl->cpuCount == 1)
       spl->cpus = xRealloc(spl->cpus, sizeof(CPUData));
-   } else {
+   else
       spl->cpus = xRealloc(spl->cpus, (pl->cpuCount + 1) * sizeof(CPUData));
```

Comment:
Why special-case this allocation based on CPU count?

Reply:
AIUI, the reason the code is this way is that when we have two or more CPUs on a system we always allocate an extra row in the array for the aggregated values (summed across all CPUs).

This behaviour is not changed here - I just added some missing error handling.

Reply:
I figured as much. Just wondered if this special-casing is really necessary or if we just say to allocate this extra entry regardless of number of CPUs …

Reply:
Yep, the special casing could certainly be removed and we'd waste that little bit of memory - *shrug*.  Since the code already caters for it, I'd be inclined to leave it as-is though.